docker/Dockerfile.api

```diff
@@ -2,12 +2,23 @@ ARG base_image_tag
 
 FROM deepset/haystack:${base_image_tag}
 
-# Create a folder for the /file-upload API endpoint with write permissions
-RUN mkdir -p /opt/file-upload && chmod 777 /opt/file-upload
+ENV SERVICE_NAME="gunicorn-service"
+
+RUN addgroup --gid 1001 $SERVICE_NAME && \
+    adduser --gid 1001 --uid 1001 --shell /bin/false --disabled-password $SERVICE_NAME && \
+    mkdir -p /var/log/$SERVICE_NAME && \
+    chown $SERVICE_NAME:$SERVICE_NAME /var/log/$SERVICE_NAME
+
+
+# Create a folder for the /file-upload API endpoint with write permissions for the service user only
+RUN mkdir -p /opt/file-upload && chown $SERVICE_NAME:$SERVICE_NAME /opt/file-upload && chmod 600 /opt/file-upload
 
 # Tell rest_api which folder to use for uploads
 ENV FILE_UPLOAD_PATH="/opt/file-upload"
 
+ENV TIKA_LOG_PATH="/var/log/$SERVICE_NAME/"
```

Comment:
From tika [docs](https://github.com/chrismattmann/tika-python/blob/master/README.md#environment-variables):

> TIKA_LOG_FILE - set the filename for the log file. default: tika.log. if it is an empty string (''), no log file is created.

Can we try this approach instead? Besides permissions issue, I'd feel more comfortable if we avoided logging into the container :)

Reply:
They made that change after releasing the last version. See the code [change](https://github.com/chrismattmann/tika-python/commit/af037d1d56a77ec96b4f46ca64c6cc0b86b5b606) and the [releases](https://github.com/chrismattmann/tika-python/releases)

Reply:
I tried it now to double-check check and it doesn't work unfortunately. Does it make sense to release the new internal release for this change? LMK

Reply:
Ok that's weird, it seems to work for me, this is the Dockerfile I'm using:
```
ARG base_image_tag

FROM deepset/haystack:${base_image_tag}

ENV SERVICE_NAME="gunicorn-service"

RUN addgroup --gid 1001 $SERVICE_NAME && \
    adduser --gid 1001 --uid 1001 --shell /bin/false --disabled-password $SERVICE_NAME && \
    mkdir -p /var/log/$SERVICE_NAME


# Create a folder for the /file-upload API endpoint with write permissions for the service user only
RUN mkdir -p /opt/file-upload && chown $SERVICE_NAME:$SERVICE_NAME /opt/file-upload && chmod 600 /opt/file-upload

# Tell rest_api which folder to use for uploads
ENV FILE_UPLOAD_PATH="/opt/file-upload"

ENV TIKA_LOG_FILE=""

EXPOSE 8000

USER $SERVICE_NAME
CMD ["gunicorn", "rest_api.application:app",  "-b", "0.0.0.0", "-k", "uvicorn.workers.UvicornWorker", "--workers", "1", "--timeout", "180"]
```

Reply:
Ok, let me try again. It could be that you built from the old branch @masci Let's both double check

Reply:
```
ack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/rest_api/application.py", line 4, in <module>
haystack-haystack-api-1   |     from rest_api.utils import get_app, get_pipelines
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/rest_api/utils.py", line 5, in <module>
haystack-haystack-api-1   |     from haystack import __version__ as haystack_version
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/__init__.py", line 21, in <module>
haystack-haystack-api-1   |     from haystack.nodes.base import BaseComponent
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/nodes/__init__.py", line 1, in <module>
haystack-haystack-api-1   |     from haystack.utils.import_utils import safe_import
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/utils/__init__.py", line 2, in <module>
haystack-haystack-api-1   |     from haystack.utils.preprocessing import convert_files_to_docs, tika_convert_files_to_docs
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/utils/preprocessing.py", line 7, in <module>
haystack-haystack-api-1   |     from haystack.nodes.file_converter import BaseConverter, DocxToTextConverter, PDFToTextConverter, TextConverter
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/nodes/file_converter/__init__.py", line 6, in <module>
haystack-haystack-api-1   |     from haystack.nodes.file_converter.tika import TikaConverter, TikaXHTMLParser
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/haystack/nodes/file_converter/tika.py", line 12, in <module>
haystack-haystack-api-1   |     from tika import parser as tikaparser
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/tika/parser.py", line 19, in <module>
haystack-haystack-api-1   |     from .tika import parse1, callServer, ServerEndpoint
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.9/site-packages/tika/tika.py", line 155, in <module>
haystack-haystack-api-1   |     fileHandler = logging.FileHandler(log_file)
haystack-haystack-api-1   |   File "/usr/lib/python3.9/logging/__init__.py", line 1146, in __init__
haystack-haystack-api-1   |     StreamHandler.__init__(self, self._open())
haystack-haystack-api-1   |   File "/usr/lib/python3.9/logging/__init__.py", line 1175, in _open
haystack-haystack-api-1   |     return open(self.baseFilename, self.mode, encoding=self.encoding,
haystack-haystack-api-1   | PermissionError: [Errno 13] Permission denied: '/tmp/tika.log'
haystack-haystack-api-1   | [2022-12-23 13:08:37 +0000] [28] [INFO] Worker exiting (pid: 28)
haystack-haystack-api-1   | [2022-12-23 13:08:38 +0000] [1] [INFO] Shutting down: Master
haystack-haystack-api-1   | [2022-12-23 13:08:38 +0000] [1] [INFO] Reason: Worker failed to boot. ```

Reply:
As soon as I set `ENV TIKA_LOG_PATH="/var/log/$SERVICE_NAME/"` it works ok

Reply:
I think you're mixing `TIKA_LOG_FILE` and `TIKA_LOG_PATH`? Can you try out my Dockerfile I posted above?

Reply:
I checked and triple-checked, your version and mine. Let's do a building session together today or next week.

Reply:
Are you maybe installing Tika from sources?

Reply:
Maybe a caching problem I have? I'm cleaning all the local builds and start from scratch, I'll keep you posted

Reply:
I've just built everything from scratch using the Dockerfile I pasted in this comment thread:
1. `docker buildx bake base-cpu --set "*.platform=linux/arm64" --no-cache`
2. I've changed the default `docker-compose.yaml` with `image: "deepset/haystack:cpu-local"` under the `haystack-api` service
3. `docker-compose up`
and the demo seems to work fine

Reply:
@masci From your steps, it seems that you're only rebuilding the `base-cpu` image, not the `cpu` image. If I follow your steps with `docker buildx bake cpu --set "*.platform=linux/arm64" --no-cache`, I get the following error:
```
haystack-haystack-api-1   | [2022-12-27 11:00:02 +0000] [6] [ERROR] Exception in worker process
haystack-haystack-api-1   | Traceback (most recent call last):
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
haystack-haystack-api-1   |     worker.init_process()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/uvicorn/workers.py", line 66, in init_process
haystack-haystack-api-1   |     super(UvicornWorker, self).init_process()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/workers/base.py", line 134, in init_process
haystack-haystack-api-1   |     self.load_wsgi()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
haystack-haystack-api-1   |     self.wsgi = self.app.wsgi()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/app/base.py", line 67, in wsgi
haystack-haystack-api-1   |     self.callable = self.load()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
haystack-haystack-api-1   |     return self.load_wsgiapp()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
haystack-haystack-api-1   |     return util.import_app(self.app_uri)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/gunicorn/util.py", line 359, in import_app
haystack-haystack-api-1   |     mod = importlib.import_module(module)
haystack-haystack-api-1   |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
haystack-haystack-api-1   |     return _bootstrap._gcd_import(name[level:], package, level)
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
haystack-haystack-api-1   |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/rest_api/application.py", line 13, in <module>
haystack-haystack-api-1   |     app = get_app()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/rest_api/utils.py", line 28, in get_app
haystack-haystack-api-1   |     from rest_api.controller import file_upload, search, feedback, document, health
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/rest_api/controller/file_upload.py", line 20, in <module>
haystack-haystack-api-1   |     indexing_pipeline: Pipeline = get_pipelines().get("indexing_pipeline", None)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/rest_api/utils.py", line 59, in get_pipelines
haystack-haystack-api-1   |     pipelines = setup_pipelines()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/rest_api/pipeline/__init__.py", line 28, in setup_pipelines
haystack-haystack-api-1   |     query_pipeline = Pipeline.load_from_yaml(Path(config.PIPELINE_YAML_PATH), pipeline_name=config.QUERY_PIPELINE_NAME)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/pipelines/base.py", line 1908, in load_from_yaml
haystack-haystack-api-1   |     return cls.load_from_config(
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/pipelines/base.py", line 1969, in load_from_config
haystack-haystack-api-1   |     validate_config(pipeline_config, strict_version_check=strict_version_check)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/pipelines/config.py", line 245, in validate_config
haystack-haystack-api-1   |     validate_schema(pipeline_config=pipeline_config, strict_version_check=strict_version_check, extras=extras)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/pipelines/config.py", line 302, in validate_schema
haystack-haystack-api-1   |     schema = load_schema()
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/nodes/_json_schema.py", line 423, in load_schema
haystack-haystack-api-1   |     raise e
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/nodes/_json_schema.py", line 417, in load_schema
haystack-haystack-api-1   |     update_json_schema(main_only=True)
haystack-haystack-api-1   |   File "/opt/venv/lib/python3.10/site-packages/haystack/nodes/_json_schema.py", line 437, in update_json_schema
haystack-haystack-api-1   |     os.makedirs(destination_path, exist_ok=True)
haystack-haystack-api-1   |   File "/usr/local/lib/python3.10/os.py", line 225, in makedirs
haystack-haystack-api-1   |     mkdir(name, mode)
haystack-haystack-api-1   | PermissionError: [Errno 13] Permission denied: '/opt/venv/lib/python3.10/site-packages/haystack/json-schemas'
haystack-haystack-api-1   | [2022-12-27 11:00:02 +0000] [6] [INFO] Worker exiting (pid: 6)
haystack-haystack-api-1   | [2022-12-27 11:00:18 +0000] [1] [INFO] Shutting down: Master
haystack-haystack-api-1   | [2022-12-27 11:00:18 +0000] [1] [INFO] Reason: Worker failed to boot.
```

Reply:
@bogdankostic my bad I pasted the wrong command, that's a different error - incidentally, while debugging the above I was able to reproduce the tika problem finally 😅